matrix-editor/lib/normalize.js

```diff
@@ -1,8 +1,8 @@
-const Sanscript = window.Sanscript;
+import {Sanscript} from "./sanscript";
```

Comment:
This way we avoid polluting the `window` object, considered a bad practice [^1] and can cause simple bugs and performance issues [^2].

Instead of use [ES6 modules](https://hacks.mozilla.org/2015/08/es6-in-depth-modules/) that can be handled by JS tools & browser.

[^1]: https://dev.to/efpage/how-to-avoid-namespace-pollution-in-javascript-3h3
[^2]: https://stackoverflow.com/a/12042905

Reply:
Thanks, this is a good reminder that my Javascript style is 10 years out of date...

I'm starting to turn the `*.js` files in the `/lib` directory into ES6 modules now. I like the idea that I can run/test everything without a compile step. The only problem I'm having is that most JS now is distributed as UMD modules... do you have any suggestions for dealing with UMD in an ES6 context? 

For example, the Raphäel library that you mention below — if I install it with `yarn`, how could I go about using it as an ES6 module? Or should I just give in and succumb to the CommonJS/Webpack hegemony?

Reply:
>Thanks, this is a good reminder that my Javascript style is 10 years out of date...

I wrote JS the exact same way until a few years ago when I **had to** since we were using Vue at $work. Even my Open Source work was a bit old-school, and I'm trying to [catch up now](https://github.com/jenkinsci/active-choices-plugin/pull/57). Nothing wrong with that :)

>I'm starting to turn the *.js files in the /lib directory into ES6 modules now. I like the idea that I can run/test everything without a compile step. The only problem I'm having is that most JS now is distributed as UMD modules... do you have any suggestions for dealing with UMD in an ES6 context?

I never had to deal with that, but looks like others [tried before](https://2ality.com/2017/04/setting-up-multi-platform-packages.html) with success to use UMD. It looks to require a bit more of work…

>For example, the Raphäel library that you mention below — if I install it with yarn, how could I go about using it as an ES6 module?

Assuming you are using ES6 modules, then you should be able to:

```js
import {Raphael} from 'raphael';
```

and then use it in your code.

>Or should I just give in and succumb to the CommonJS/Webpack hegemony?

:smile: I think you should try to use the approach you are most comfortable with, for both writing the code for `matrix-editor` and for maintaining it (e.g. future contributions from other users). I think the only important things are to have the dependencies somewhere like `package.json`, and unit tests :+1: